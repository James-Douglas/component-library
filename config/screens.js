const screens = {
  xs: "375px",
  sm: "720px",
  md: "1200px",
  lg: "1400px",
  xl: "1900px",
  xxl: "2560px"
};

export default screens;
