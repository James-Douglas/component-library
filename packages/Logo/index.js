import Logo from './Logo.component';
import CTMLogo from './CTMLogo.component';

export { Logo, CTMLogo };
