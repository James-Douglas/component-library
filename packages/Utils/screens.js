const screens = {
  xs: '375px',
  sm: '768px',
  md: '1024px',
  lg: '1200px',
  xl: '1440px',
};

export default screens;
