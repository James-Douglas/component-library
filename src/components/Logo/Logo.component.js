import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import Picture, { picturePropTypes } from '../Picture/Picture.component';

const StyledLogo = styled.div`
  transition: all 200ms ease;

  ${({ theme, size }) => size === 'large' && css`
    height: ${theme.logo.heightLarge}; /* 44px */
  `}

  ${({ theme, size }) => size === 'small' && css`
    height: ${theme.logo.heightSmall}; /* 32px */
  `}
`;

const Logo = ({ size, link, picture }) => {
  const {
    src, srcsets, alt, title, className,
  } = picture;
  return (
    <StyledLogo size={size} id="logo">
      <a href={link}>
        <Picture
          src={src}
          srcsets={srcsets}
          alt={alt}
          title={title}
          className={className}
        />
      </a>
    </StyledLogo>
  );
};

Logo.propTypes = {
  /**
   * Address for the <a> tag wrapping the logo image.
   */
  link: PropTypes.string,
  /**
   * Size of the logo (small = 3.2rem, large = 4.4rem)
   */
  size: PropTypes.oneOf(['small', 'large']),
  picture: PropTypes.shape(picturePropTypes).isRequired,
};

Logo.defaultProps = {
  link: 'https://www.comparethemarket.com.au',
  size: 'large',
};

export default Logo;
