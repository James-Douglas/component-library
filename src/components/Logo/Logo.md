# Icon component
Basic logo component

## Usage

~~~js
<Logo 
  size={'w-64'}
  src={'https://www.comparethemarket.com.au/logo.svg'}
  srcsets={[{
    srcset: 'https://www.something.com/placeholder.svg',
    media: '(min-width: 300px'
  }]}
/>
~~~

## Props

1. `size` *(optional) string*
    * A TailwindCSS class (see tailwind.config.js) - typically a width or height class
2. `src` *(optional) string*
	* src for the image
3. `srcsets` *(optional) array*
	* Array of objects of { srcset, media }
