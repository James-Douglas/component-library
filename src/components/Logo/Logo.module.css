.logo {
  transition: all 200ms ease;
}

.logo.large {
  height: 4.4rem;     /* 44px */
}

.logo.small {
  height: 3.2rem;     /* 32px */
}
