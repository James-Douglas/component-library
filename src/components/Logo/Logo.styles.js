import styled, { css } from 'styled-components';

const StyledLogo = styled.div`
  transition: all 200ms ease;

  ${({ theme, size }) => size === 'large'
    && css`
      height: ${theme.logo.heightLarge}; /* 44px */
    `}

  ${({ theme, size }) => size === 'small'
    && css`
      height: ${theme.logo.heightSmall}; /* 32px */
    `}
`;

export default StyledLogo;
