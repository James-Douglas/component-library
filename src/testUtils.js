// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
