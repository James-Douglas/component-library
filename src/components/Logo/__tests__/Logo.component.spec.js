import React from 'react';
import { render } from '../../../testUtils';
import Logo from '../Logo.component';
import theme from '../../../themes/ctm.theme';
import CTMLogoInline from '../../../images/ctm-logo-inline.svg';
import CTMLogoStacked from '../../../images/ctm-logo-stacked.svg';
import CTMLogoMobile from '../../../images/ctm-logo-mobile.svg';
import screens from '../../../../config/screens';

describe('Logo', () => {
  const pictureProps = {
    src: CTMLogoMobile,
    srcsets: [
      {
        srcset: CTMLogoMobile,
        media: `(max-width: ${screens.xs}`,
      },
      {
        srcset: CTMLogoStacked,
        media: `(max-width: ${screens.lg}`,
      },
      {
        srcset: CTMLogoInline,
      },
    ],
  };

  it('renders correctly with minimal props', () => {
    const { container } = render(<Logo picture={pictureProps} />);
    expect(container.innerHTML).toMatchSnapshot();

    const a = container.querySelector('a');
    expect(a).toBeDefined();
    expect(a.getAttribute('href')).toBe('https://www.comparethemarket.com.au');

    const sources = container.querySelectorAll('source');
    expect(sources).toBeDefined();
    expect(sources[0].getAttribute('srcset')).toEqual('ctm-logo-mobile.svg');
    expect(sources[1].getAttribute('srcset')).toEqual('ctm-logo-stacked.svg');
    expect(sources[2].getAttribute('srcset')).toEqual('ctm-logo-inline.svg');
  });

  it('renders a small logo', () => {
    const { container } = render(<Logo size="small" picture={pictureProps} />);
    expect(container.querySelector('#logo')).toHaveStyle(`height: ${theme.logo.heightSmall}`);
  });
});
