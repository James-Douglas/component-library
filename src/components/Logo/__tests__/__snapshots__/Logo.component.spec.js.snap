// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly with minimal props 1`] = `"<div id=\\"logo\\" class=\\"sc-AxiKw hJQgTV\\"><a href=\\"https://www.comparethemarket.com.au\\"><picture class=\\"sc-AxjAm kaRYdH\\"><source srcset=\\"ctm-logo-mobile.svg\\" media=\\"(max-width: 375px\\"><source srcset=\\"ctm-logo-stacked.svg\\" media=\\"(max-width: 1400px\\"><source srcset=\\"ctm-logo-inline.svg\\"><img src=\\"ctm-logo-mobile.svg\\" alt=\\"\\" title=\\"\\" class=\\"sc-AxirZ iemiCJ\\"></picture></a></div>"`;
